DEVELOPER_GUIDE.md

```diff
@@ -0,0 +1,66 @@
+# Developer Guide
+
+So you want to contribute code to OpenSearch Go Client?  Excellent!  We're glad you're here.  Here's what you need to do:
+
+## Getting Started
+
+### Git Clone OpenSearch Go Client Repository
+
+Fork [opensearch-project/opensearch-go](https://github.com/opensearch-project/opensearch-go) and clone locally,
+e.g. `git clone https://github.com/[your username]/opensearch-go.git`.
+
+### Install Prerequisites
+
+#### Go 1.11
+OpenSearch Go Client builds using [Go](https://golang.org/doc/install) 1.11 at a minimum.
+
+#### Docker
+
+[Docker](https://docs.docker.com/install/) is required for building some OpenSearch artifacts and executing integration tests.
+
+### Unit Testing
+Go has a simple tool for running tests, and we simplified further by creating make command, use this command:
+ ```
+make test-unit
+```
+ 
+However, often when writing tests, you may want to run your new test as below
+```
+cd folder-path/to/test;
+go test -v -run TestName; 
+```
+
+### Integration Testing
+In order to test opensearch-go client, you need a running OpenSearch cluster. You can use Docker to accomplish this. 
+The [Docker Compose file](.ci/opensearch/docker-compose.yml) supports the ability to run integration tests for the project in local environments respectively.
+If you have not installed docker-compose, you can install it from this [link](https://docs.docker.com/compose/install/).
+
+Integration tests are often slower, so you may want to only run them after the unit test. In order to differentiate unit tests
+from integration tests, Go has a built-in mechanism for allowing you to logically separate your tests
```

Comment:
Is there a link for these build tags docs?

Reply:
There is no link to go documentation. This is more like technique used to run integration tests

Reply:
How can I learn more about what a build tag is?

Reply:
Logic is similar to build tags: https://pkg.go.dev/cmd/go#hdr-Build_constraints

Reply:
Right, link to that on reference to build tags

Reply:
Ack

Reply:
@jmazanec15 can you take another look? Thanks.